server/lib/resolvers/builders/knex.js

```diff
@@ -0,0 +1,60 @@
+const { VM } = require('vm2')
+const { auditLog, log } = require('../../../lib/util/logger')
+const { updateResolverMetrics } = require('../../../metrics')
+const { isEmpty } = require('lodash')
+
+function buildKnexResolver (dataSource, compiledRequestMapping, compiledResponseMapping) {
+  return function resolve (obj, args, context, info) {
+    return new Promise((resolve, reject) => {
+      const dataSourceClient = dataSource.getClient()
+      const requestTimeStart = Date.now()
+      info['dataSourceType'] = dataSource.type
+
+      const sandbox = {
+        db: dataSourceClient,
+        arguments: args,
+        parent: obj
+      }
+
+      const vm = new VM({
+        sandbox
+      })
+
+      // result is whatever is implicitly returned from the script
+      // in this case it will be the query builder object from knex
+      const query = vm.run(compiledRequestMapping)
+
+      // this is a super rough implementation
+      query.then((result) => {
```

Comment:
[Security] Does split between request/response still makes sense when we do not use sql?
With this split we have one operation outside sandbox (promise with query result), but if users return object different than promise with custom `then` method this method will be called outside sandbox.

I guess it depends if we want to really support kinda SaaS model here and it's not actual issue. Just wanted to point out.  

TL;DR - we may revisit request response format later.